stumpy/stump_tiles.py

```diff
@@ -0,0 +1,780 @@
+# STUMPY
+# Copyright 2019 TD Ameritrade. Released under the terms of the 3-Clause BSD license.
+# STUMPY is a trademark of TD Ameritrade IP Company, Inc. All rights reserved.
+
+import logging
+
+import numpy as np
+from numba import njit, prange
+import numba
+
+from . import core, config
+from .aamp import aamp
+
+logger = logging.getLogger(__name__)
+
+
+@njit(
+    # "(f8[:], f8[:], i8, f8[:], f8[:], f8[:], f8[:], f8[:], f8[:], f8[:], f8[:],"
+    # "b1[:], b1[:], b1[:], b1[:], i8[:, :, :], i8[:, :], i8, f8[:, :, :], f8[:, :],
+    # f8[:, :], i8[:, :, :], i8[:, :], i8[:, :], b1)",
+    fastmath=True,
+)
+def _compute_tiles(
```

Comment:
This function seems very long.Can we break this up into `_compute_tiles` and `_compute_diagonals`, where the former (`_compute_tiles`) handles figuring out the tile info and then calls the latter (`_compute_diagonals`) to compute the distances?

Reply:
Yes but isn't computing along rows of diagonals when the `STUMPY_N_DIAGONALS = 1` basically the same thing as `stump._compute_diagonals`?

Reply:
Correct, but that doesn't work when `STUMPY_N_DIAGONALS != ``

Reply:
> but that doesn't work when STUMPY_N_DIAGONALS != 1

I don't understand what you mean here. I'm basically saying that we could, with some work, have a single `_compute_diagonals` that would work for both `stump` (set `STUMPY_N_DIAGONALS = 1` and `STUMPY_MAX_TILE_SIZE = np.inf`) and `stump_tiles` (use default params)

Reply:
Never mind. I was wrong